pkg/scalers/azure_blob_scaler.go

```diff
@@ -42,7 +44,7 @@ type azureBlobMetadata struct {
 var azureBlobLog = logf.Log.WithName("azure_blob_scaler")
 
 // NewAzureBlobScaler creates a new azureBlobScaler
-func NewAzureBlobScaler(config *ScalerConfig) (Scaler, error) {
+func NewAzureBlobScaler(httpClient *http.Client, config *ScalerConfig) (Scaler, error) {
```

Comment:
Changing the contract for creating a new scaler seems unnecessary. I very much like the recent changes where we simplified the creation of a scaler object with ScalerConfig. It's a cleaner approach.